assets/training/model_management/src/azureml/model/mgmt/utils/common_utils.py

```diff
@@ -121,6 +126,60 @@ def create_namespace_from_dict(var: Any):
     return var
 
 
+def get_dict_from_comma_separated_str(dict_str: str, item_sep: str, kv_sep: str, do_eval: bool = False) -> Dict:
+    """Create and return dictionary from string.
+
+    :param dict_str: string to be parsed for creating dictionary
+    :type dict_str: str
+    :param item_sep: char separator used for item separation
+    :type item_sep: str
+    :param kv_sep: char separator used for key-value separation. Must be different from item separator
+    :type kv_sep: str
+    :param do_eval: Whether to eval parsed value string. Default is False
+    :type do_eval: bool
+    :return: Resultant dictionary
+    :rtype: Dict
+    """
+    if not dict_str:
+        return {}
+    item_sep = item_sep.strip()
+    kv_sep = kv_sep.strip()
+    if len(item_sep) > 1 or len(kv_sep) > 1:
+        raise Exception("Provide single char as separator")
+    if item_sep == kv_sep:
+        raise Exception("item_sep and kv_sep are equal.")
+    parsed_dict = {}
+    kv_pairs = dict_str.split(item_sep)
+    for item in kv_pairs:
+        split = item.split(kv_sep)
+        if len(split) == 2:
+            key = split[0].strip()
+            val = split[1].strip()
+            if do_eval:
+                try:
+                    val = eval(split[1].strip())
```

Comment:
Avoid using eval unless input can be trusted.

Reply:
@elesel understood the concern. Need eval for class names that would be provided in input. Is there a better way to do this? Let me also take a look

Reply:
one ex: 

`trust_remote_code=True, torch_dtype': 'torch.bfloat16', 'low_cpu_mem_usage': True`

Reply:
Removed `eval`. There is still use case to eval. But will validate that in separate PR. thanks for highlighting security concerns with eval.